### ISSUE TITLE
Boxing information (and possibly more) is lost during Java->Kotlin inheritance

### ISSUE THREAD

**Issue by owengray-google** [Kotlin/dokka#2996]
**Describe the bug**
In a `JavaClass`, have
```
public int unboxed = 5;
public Integer boxed = 5;
public int[] arrayOfUnboxed = {};
public Integer[] arrayOfBoxed = {};
```
Have `public class KotlinClass : JavaClass`.
Generate as-Java docs for `KotlinClass`.

**Expected behaviour**
The `boxed` fields use `Integer`s and the `unboxed` fields use `int`s.
**Actual behavior**
All fields use `Integer`s. 
The original boxing information is lost when Java types are converted to Kotlin (during inheritance?).

**Screenshots**
<img width="625" alt="Screenshot 2023-05-15 at 10 06 59 AM" src="https://github.com/Kotlin/dokka/assets/56644404/0e2ff9e5-1b36-4246-a9a5-dfcd24051330">

(Dokka version `1.8.10-dev-203`)

This is likely something that should be addressed at the same time as https://github.com/Kotlin/dokka/issues/2603 (i.e. as part of the architectural changes necessary to fix that bug).

**Comment by owengray-google** [comment]
I don't think this is related to the `kotlin-as-java` plugin.

The reason that the boxing information is missing from the as-Java docs is because the boxing information is not present in the model, long before the plugin is applied. I think this is an issue with the model.

**Comment by IgnatBeresnev** [comment]
Doesn't it affect the output of the `kotlin-as-java` plugin? `Integer` -> `kotlin.Int` -> `int`

**Comment by IgnatBeresnev** [comment]
Yeah, good point. I think it'll depend on how the first transition is handled

I'm afraid that if we parse `java.lang.Integer` and pass it into the documentable model as `GenericTypeConstructor(java.lang.Integer)`, then Dokka's own HTML plugin (without KaJ) will create external links to Java's Javadoc pages for `Integer` (or even worse - use it as a type, like `val i: Integer`) - I don't think such behaviour is desirable

However, if the original type is somehow preserved in the documentable model, but not used anywhere directly, we'll need to adjust the `kotlin-as-java` plugin accordingly, yeah 👍
